src/main/java6/net/finmath/time/Schedule.java

```diff
@@ -116,19 +116,50 @@ public double getPeriodLength(int periodIndex) {
 		return periodLength[periodIndex];
 	}
 
-	@Override
-	public String toString() {
-		return "Schedule [referenceDate=" + referenceDate + ", periods="
-				+ periods + ", daycountconvention=" + daycountconvention + "]";
-	}
-
-
 	/* (non-Javadoc)
 	 * @see java.lang.Iterable#iterator()
 	 */
 	@Override
 	public Iterator<Period> iterator() {
 		return periods.iterator();
 	}
-
+	
+	/**
+	 * Creates a date by adding a double dateOffset (Act/365) to a reference date
+	 * 
+	 * @param referenceDate reference date to add dateOffset to
+	 * @param dateOffset dateOffset stored as a double with Act/365 convention
+	 * @return The date resulting from adding a double dateOffset (Act/365) to a reference date
+	 */
+	public static LocalDate getDateFromDouble(LocalDate referenceDate, double dateOffset) {
+		if(!(internalDayCounting instanceof DayCountConvention_ACT_365))
+			throw new IllegalArgumentException("Method expects ACT/365 as internalDayCounting, not " + internalDayCounting);
+		int numberOfDays = (int)(dateOffset*365);
```

Comment:
This is dangerous and I believe it is also a bug. It is also not consistent with the formulas used at other places.

The rounding may lead to loosing a full day. I belive you can easily construct examples (3 days may be an example). This bug was there before and fixed via
`getReferenceDate().plusDays((int)Math.round(dateOffset*365))`

Just check you code with 1, 2, 3, 4, 5 days - (I belive that 3 days will lead to losing 1 day).

See also the method getDateFromModelTime in ForwardCurveNelsonSiegelSvennson:

```
	private LocalDate getDateFromModelTime(double fixingTime) {
		return getReferenceDate().plusDays((int)Math.round(fixingTime*365.0));
	}

```

I would change this implementation.